server/controllers/medical/patients/index.js

```diff
@@ -284,137 +286,82 @@ function hospitalNumberExists(req, res, next) {
-
-  // use MYSQL to look up the reference
-  // TODO This could probably be optimized
-  sql =
-    `SELECT q.uuid, q.project_id, q.debtor_uuid, q.first_name, q.last_name, q.middle_name,
-      q.sex, q.dob, q.origin_location_id, q.reference, q.text,
-      q.account_id, q.price_list_uuid, q.is_convention, q.locked
-    FROM (
-      SELECT p.uuid, p.project_id, p.debtor_uuid, p.first_name, p.last_name, p.middle_name,
-      p.sex, p.dob, CONCAT(proj.abbr, p.reference) AS reference, p.origin_location_id, d.text,
-      dg.account_id, dg.price_list_uuid, dg.is_convention, dg.locked
-      FROM patient AS p JOIN project AS proj JOIN debtor AS d JOIN debtor_group AS dg
-        ON p.debtor_uuid = d.uuid AND d.group_uuid = dg.uuid AND p.project_id = proj.id
-    ) AS q
-    WHERE q.reference = ?;`;
-
-  db.exec(sql, [reference])
-  .then(function (rows) {
+function visit(req, res, next) {
+  var visitData = req.body;
 
-    if (_.isEmpty(rows)) {
-      res.status(404).send();
-    } else {
-      res.status(200).json(rows[0]);
-    }
+  logVisit(visitData, req.session.user.id)
+    .then(function (result) {
 
-  })
-  .catch(next)
-  .done();
+      // Assign patient ID as confirmation
+      result.uuid = visitData.uuid;
 
+      res.status(200).send(result);
+    })
+    .catch(next)
+    .done();
 }
 
 /**
-* GET /patient/search/fuzzy/:match
-* @desc Performs fuzzy searching on patient names
-*/
-function searchFuzzy(req, res, next) {
-
-  var sql, match = req.params.match;
-
-  if (!match) { next(new Error('No parameter provided!')); }
-
-  // search on the match parameter
-  sql =
-    `SELECT BUID(p.uuid) as uuid, p.project_id, BUID(p.debtor_uuid) AS debtor_uuid, p.first_name,
-      p.last_name,  p.middle_name, p.sex, p.dob, p.origin_location_id, p.reference, proj.abbr, d.text,
-      dg.account_id, BUID(dg.price_list_uuid) as price_list_uuid, dg.is_convention, dg.locked
-    FROM patient AS p JOIN project AS proj JOIN debtor AS d JOIN debtor_group AS dg
-    ON p.debtor_uuid = d.uuid AND d.group_uuid = dg.uuid AND p.project_id = proj.id
-    WHERE
-      LEFT(LOWER(CONCAT(p.last_name, \' \', p.middle_name, \' \', p.first_name )), CHAR_LENGTH(?)) = ? OR
-      LEFT(LOWER(CONCAT(p.last_name, \' \', p.first_name, \' \', p.middle_name)), CHAR_LENGTH(?)) = ? OR
-      LEFT(LOWER(CONCAT(p.first_name, \' \', p.middle_name, \' \', p.last_name)), CHAR_LENGTH(?)) = ? OR
-      LEFT(LOWER(CONCAT(p.first_name, \' \', p.last_name, \' \', p.middle_name)), CHAR_LENGTH(?)) = ? OR
-      LEFT(LOWER(CONCAT(p.middle_name, \' \', p.last_name, \' \', p.first_name)), CHAR_LENGTH(?)) = ? OR
-      LEFT(LOWER(CONCAT(p.middle_name, \' \', p.first_name, \' \', p.last_name)), CHAR_LENGTH(?)) = ?
-    LIMIT 10;`;
-
-  // man. That's a lot of matches
-  db.exec(sql, [match, match, match, match, match, match, match, match, match, match, match, match])
-  .then(function (rows) {
-    res.status(200).json(rows);
-  })
-  .catch(next)
-  .done();
+ * @function logVisit
```

Comment:
Gotcha.  I was a bit confused in the changelog about what happened and favored preserving old code to removing it.  I'll remove it now.